compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/inlineclasses/InlineClassAbi.kt

```diff
@@ -56,6 +57,13 @@ object InlineClassAbi {
      * to avoid clashes between overloaded methods.
      */
     fun mangledNameFor(irFunction: IrFunction): Name {
+        val suffix = when {
```

Comment:
@ilmirus Could you take a look there is similar code that is used in MemoizedInlineClassReplacements from https://github.com/JetBrains/kotlin/commit/7a096b305f11bece9e39ffd8875a766db5df677a

Reply:
There's still a problem with this commit. The methods of the result class have to be mangled (e.g. the compiled function is called `Result.getOrNull-impl` instead of `Result.getOrNull`). Right now it just disables mangling entirely. On the other hand, my patch did replace functions unnecessarily in some cases. I've updated the PR to fix both problems.